lib/utils.c

```diff
@@ -39,6 +39,54 @@
 
 int timestamp_short;
 
+int read_prop(char *dev, char *prop, long *value)
+{
+	char fname[41], buf[80], *endp;
+	ssize_t len;
+	FILE *fp;
+	long result;
+
+	sprintf(fname, "/sys/class/net/%s/%s", dev, prop);
+	fp = fopen(fname, "r");
+	if (fp == NULL) {
+		if (strcasecmp(prop, "tun_flags"))
+			fprintf(stderr, "open %s: %s\n", fname,
+				strerror(errno));
+		return -1;
+	}
+
+	len = fread(buf, sizeof(buf), 1, fp);
```

Comment:
The return value of `fread()` is different than `read()` -- try reading the manual page and see if you can come up with the correct way to treat the return value. Note that the manual specifies that we need to use `feof()` and `ferror()` to get it right:

       On  success,  fread()  and  fwrite() return the number of items read or
       written.  This number equals the number of bytes transferred only  when
       size  is 1.  If an error occurs, or the end of the file is reached, the
       return value is a short item count (or zero).

       fread() does not distinguish between end-of-file and error, and callers
       must use feof(3) and ferror(3) to determine which occurred.

It may help to look for uses of `fread()` in this repository or in the kernel or any other codebase to see if you can get it right.